library/alloc/src/vec/mod.rs

```diff
@@ -1231,6 +1237,16 @@ impl<T, A: Allocator> Vec<T, A> {
     /// }
     /// ```
     ///
+    /// The aliasing guarantee works out this way:
+    ///
+    /// ```rust
+    /// let mut v = vec![0];
+    /// let ptr = v.as_ptr();
+    /// let x = ptr.read();
+    /// v[0] = 5;
+    /// // Notably, the write above did *not* invalidate `ptr1`:
```

Comment:
(hi, bystander's nit): i think `ptr1` here should actually be `ptr`?

Reply:
yeah this isn't landable code currently, it's just an example for communicating with the other folks as we workshop things